nemo_skills/pipeline/summarize_results.py

```diff
@@ -105,15 +105,51 @@ def summarize_results(
         results_dir = Path(temp_dir) / Path(results_dir).name
 
     # running compute_metrics.py to get greedy, majority and pass @k results for all benchmarks available
-    # Check if there is an eval-results dir inside the results_dir
-    eval_results_dir = Path(results_dir) / 'eval-results'
-    if eval_results_dir.exists() and eval_results_dir.is_dir():
-        results_dir = eval_results_dir
-    benchmarks_paths = [path for path in glob.glob(f'{results_dir}/*') if '-logs' not in os.path.basename(path)]
+
+    # Check for all possible directory structures
+    # 1. {results_dir}/eval-results/{benchmark}/output*jsonl
+    # 2. {results_dir}/{benchmark}/output*jsonl
+    # 3. {results_dir}/output*jsonl
+
+    # List to store all valid benchmarks results paths
+    benchmarks_paths = []
+
+    # Check for Option 3 - Root directory corresponds to a benchmark
+    if Path(results_dir).is_dir() and len(glob.glob(f'{results_dir}/output*jsonl')) > 0:
+        benchmarks_paths = [results_dir]
+    else:
+        cand_results_dir = Path(results_dir) / 'eval-results'
+        # Check for Option 1
+        if cand_results_dir.exists() and cand_results_dir.is_dir():
+            results_dir = cand_results_dir
+        elif len(glob.glob(f'{results_dir}/*/output*jsonl')) > 0:
```

Comment:
I think this condition is just not needed. It will fail right below when checking individual folders with a clear message anyway. But if you want to keep it, please change to == 0 and raise an error, as current `result_dir = result_dir` is redundant and a bit confusing

Reply:
Ahh, fair point. Have changed the logic now.